File: articles/ranking-filtering-and-recommendation/julia/ranking_filtering_recommendation.jl
# Synthetic ranking score example in Julia.
function ranking_score(text_match, quality, freshness, diversity_bonus, risk_penalty)
    0.36 * text_match + 0.30 * quality + 0.16 * freshness + 0.14 * diversity_bonus - 0.20 * risk_penalty
end
println(round(ranking_score(0.92, 0.88, 0.60, 0.35, 0.04), digits=6))

File: articles/ranking-filtering-and-recommendation/haskell/RankingFilteringRecommendation.hs
module Main where

rankingScore :: Double -> Double -> Double -> Double -> Double -> Double
rankingScore textMatch quality freshness diversityBonus riskPenalty =
  0.36 * textMatch + 0.30 * quality + 0.16 * freshness + 0.14 * diversityBonus - 0.20 * riskPenalty

main :: IO ()
main = print (rankingScore 0.92 0.88 0.60 0.35 0.04)
